ui_framework/components/button/button.test.js

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable no-undef */
+
+import React from 'react';
+import { render, shallow } from 'enzyme';
+import sinon from 'sinon';
+
+import {
+  BUTTON_TYPES,
+  KuiButton,
+} from './button';
+
+describe('KuiButton', () => {
+  describe('Baseline', () => {
+    test('is rendered', () => {
+      const $button = render(
+        <KuiButton />
+      );
+
+      expect($button)
+        .toMatchSnapshot();
+    });
+  });
+
+  describe('HTML attributes', () => {
```

Comment:
I would fold all of these into one single test (without a `describe`), e.g.

```js
test('renders HTML attributes as-is', () => {
})
```

or something like that. I've gradually made my snapshots tests a little bit "broader" instead of being super-focused (it's kinda "unlearning TDD"). The point is that the snapshot together with the test is communicating what's allowed. It also creates "less noise" imo.